src/Credentials/CredentialTrait.php

```diff
@@ -0,0 +1,68 @@
+<?php
+namespace Aws\Credentials;
+
+use Aws\Exception\CredentialsException;
+use GuzzleHttp\Promise;
+
+/**
+ * @internal Sourcing credential profile data from file helpers.
+ */
+trait CredentialTrait
+{
+    /**
+     * Check profile availability in a given file.
+     * Returns a string describing any error that could occur.
+     *
+     * @param string $profile
+     * @param string $filename
+     *
+     * @return null|string
+     */
+    private static function checkProfile($profile, $filename)
+    {
+        if (!is_readable($filename)) {
+            return "Cannot read credentials from $filename";
+        }
+        $data = parse_ini_file($filename, true);
+        if ($data === false) {
+            return "Invalid credentials file: $filename";
+        }
+        if (!isset($data[$profile])) {
+            return "'$profile' not found in $filename";
+        }
+    }
+
+    private static function getProfileData($profile, $filename)
```

Comment:
This method doesn't really work... Maybe checkProfile should be combined with this method so that you're only loading the ini file once.